designs/2023-rule-performance-statistics/README.md

```diff
@@ -0,0 +1,328 @@
+- Repo: eslint/eslint
+- Start Date: 2023-03-10
+- RFC PR: <https://github.com/eslint/rfcs/pull/108>
+- Authors: Mara Nikola Kiefer ([@mnkiefer](https://github.com/mnkiefer))
+
+# Performance Statistics
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This document describes a new ESLint flag `stats`, which enables the user to obtain a series of runtime *statistics* on top of their final lint results (refer to the table below).
+
+These statistics include more fine-grained timing information, such as the parse-, fix-, and lint-times and the number of directives, fix passes, and violations.
+
+| Stats          |        |       | Description |
+|----------------|--------|-------|-------------|
+| **times**      |        |       | Object containing the times spent on (parsing, fixing, linting) a file. |
+|                | passes |       | Array containing the times spent on (parsing, fixing, linting) a file for each run- or fix-pass. |
+|                |        | fix   | The time that is spent on fixing a file. |
+|                |        | parse | The time that is spent on parsing a file. |
+|                |        | rules | Object containing the times spent on each rule. |
+| **fixPasses**  |        |       | The number of times ESLint has applied at least one fix after linting. |
+| **directives** |        |       | The number of disable directives applied to silence rule violations. |
+| **violations** |        |       | The number of times a rule has been violated. |
```

Comment:
Does this include only rule violations or all lint messages (e.g., invalid inline rule configs)? Does it include rule violations suppressed with disable directives? Also, as this is a top-level property, is it a sum of violations from each pass?

Reply:
Removed violations from RFC